docs/content/guides/dagster/local-to-production-tutorial.mdx

```diff
@@ -0,0 +1,420 @@
+
+<Note>
+  At the end of this guide, you'll find an optional section that demonstrates a
+  pattern for mocking resources that can be helpful when writing unit tests.
+</Note>
+
+### Setup
+
+If you want to follow along with this guide you have two options:
+#### Clone the example code:
+This will give you the final version of the code we will write in this guide
+
+```bash
+pip install dagster, dagit
+dagster new-project --example local_to_prod
+cd local_to_prod
+pip install .
+```
+
+#### Start with a scaffolded project:
+This will create an empty project where you can copy the code as we write it in this guide
+
+```bash
+pip install dagster, dagit
+dagster new-project --quick-start local_to_prod
+cd local_to_prod
+pip install pandas pytest dagster-snowflake dagster-snowflake-pandas
+```
+
+### Part one: Local development
+
+In this section we will:
+
+- Write our assets
+- Write a resource to fetch data from Hacker News
+- Add the assets to our repository
+- Add run configuration for the Snowflake IO manager
+- Materialize assets in Dagit
+
+Let’s start by writing the asset that will contain all of the Hacker News data. We'll interact with the dataset as a Pandas DataFrame.
+
+```python file=/guides/dagster/local_to_prod_tutorial/assets.py startafter=start_items endbefore=end_items
+# assets.py
+
+
+@asset(
+    required_resource_keys={"hn_client"},
+)
+def items(context) -> pd.DataFrame:
+    """Items from the Hacker News API: each is a story or a comment on a story."""
+    hn_client = context.resources.hn_client
+
+    rows = []
+    for item_id in range(1, hn_client.fetch_max_item_id()):
+        rows.append(hn_client.fetch_item_by_id(item_id))
+
+    result = pd.DataFrame(rows, columns=hn_client.item_field_names).drop_duplicates(
+        subset=["id"]
+    )
+    result.rename(columns={"by": "user_id"}, inplace=True)
+    return result
+```
+
+In this asset we use a resource (`hn_client`) that hasn't been written yet. We'll write it in a moment. First, let's write the assets to create the `comments` and `stories` datasets:
+
+```python file=/guides/dagster/local_to_prod_tutorial/assets.py startafter=start_comments_stories endbefore=end_comments_stories
+# assets.py
+
+
+@asset
+def comments(items: pd.DataFrame) -> pd.DataFrame:
+    """Comments from the Hacker News API."""
+    return items[items["type"] == "comment"]
+
+
+@asset
+def stories(items: pd.DataFrame) -> pd.DataFrame:
+    """Stories from the Hacker News API."""
+    return items[items["type"] == "story"]
+```
+
+Now let's write the resource to fetch data from Hacker News.
+
+<Note>
+
+When you’re in the beginning development stages, you may find it easier to write the code to fetch data directly in your assets or ops. This is great for initial development, but eventually you may want to refactor your code to use resources. In this case we’re going to jump right into defining a resource because we're going to use it again in the optional section at the end of this guide.
```

Comment:
How about

```
In many cases, it is often most convenient to interact with an external service directly in assets or ops, 
rather than write a resource. We recommend that you refactor your code to use resources 
in the following cases:
* you need multiple assets or ops to interact with the service in a consistent way 
* you want to interact with a different service in certain scenarios (ie. a staging environment, or unit tests)
```

I think it still could benefit from an explanation of why we are ignoring our own advice in this guide though. So maybe an additional sentence to address that